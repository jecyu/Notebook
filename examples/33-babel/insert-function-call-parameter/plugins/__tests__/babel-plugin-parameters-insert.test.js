const pluginTester = require('babel-plugin-tester').default;

pluginTester({
    plugin: require('../babel-plugin-parameters-insert'),
    babelOptions: {
        parserOpts: {
            sourceType: 'unambiguous',
            plugins: ['jsx']
        }
    },
    tests: {
        'console.xx 前插入了 CallExpression的AST': {
            code: `
            console.log(1);

            function func() {
                console.info(2);
            }
        
            export default class Clazz {
                say() {
                    console.debug(3);
                }
                render() {
                    return <div>{console.error(4)}</div>
                }
            }
            `,
            snapshot: true
        }
    }
})