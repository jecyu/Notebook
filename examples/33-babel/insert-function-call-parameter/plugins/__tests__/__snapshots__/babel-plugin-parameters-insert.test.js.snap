// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unknown plugin console.xx 前插入了 CallExpression的AST: console.xx 前插入了 CallExpression的AST 1`] = `

console.log(1);

function func() {
    console.info(2);
}
        
export default class Clazz {
    say() {
        console.debug(3);
    }
    render() {
        return <div>{console.error(4)}</div>
    }
}

      ↓ ↓ ↓ ↓ ↓ ↓

console.log("filename: (1, 0)");
console.log(1);

function func() {
  console.log("filename: (4, 4)");
  console.info(2);
}

export default class Clazz {
  say() {
    console.log("filename: (9, 8)");
    console.debug(3);
  }

  render() {
    return <div>{[console.log("filename: (12, 21)"), console.error(4)]}</div>;
  }
}


`;
